src/js/structs/Batch.js

```diff
@@ -44,35 +44,44 @@ import TransactionTypes from '../constants/TransactionTypes';
 
 class Batch {
   constructor() {
-    const batch = [];
-
-    this.add = this.add.bind(batch);
-    this.reduce = this.reduce.bind(batch);
+    concealContext(this, []);
   }
 
   /**
    * Add an action to the batch.
    *
+   * @this Array
    * @param {Object} item
+   * @returns {Batch} Returns the new batch
    */
   add(item) {
+    // Operate on a new array
+    const batch = this.slice();
```

Comment:
:thinking: 
```
slice(): 621ms
slice(0): 633ms
```
__10,000 iterations on an array with 100,000 objects, but the tests are kind of oscillating__